Sources/Auth/Services/Wallet/AuthService.swift

```diff
@@ -1,6 +1,30 @@
 import Foundation
 
 actor AuthService {
+    enum Errors: Error {
+        case malformedPairingURI
+    }
+
+    private let appPairService: AppPairService
+
+    private let walletPairService: WalletPairService
+
+    init(appPairService: AppPairService, walletPairService: WalletPairService) {
+        self.appPairService = appPairService
+        self.walletPairService = walletPairService
+    }
+
+    func connect() async throws -> String {
+        let uri = try await appPairService.create()
+        return uri.absoluteString
+    }
+
+    func pair(uri: String) async throws {
+        guard let pairingURI = WalletConnectURI(string: uri) else {
+            throw Errors.malformedPairingURI
+        }
+        try await walletPairService.pair(pairingURI)
+    }
 
     func respond(respondParams: RespondParams) async throws {
```

Comment:
My intention was that AuthService is for wallet to call `respond(respondParams: RespondParams)` method only.
`pair` and `request` are called from client directly
`request` calls `appPairService.create()`

take a look at client api specs:
https://github.com/WalletConnect/walletconnect-specs/pull/88/files#diff-6409c08fd6475bc986e81c120780838a754491571d419553faee490bb9922801

Reply:
Moved to AuthClient